### ISSUE TITLE
Actually implement this

### ISSUE THREAD

**Issue by ikeydoherty** [solus-project/binman#1]
Long story short I'm pulling me hair out with how long each repo settle is taking. This needs implementing immediately ..

In order of Things To Make It All Work:

 - [x] Add basic `.eopkg` add support (cache into `pool` and `repo` storage)
 - [x] Add `eopkg-index.xml` write support (stable sort + constant time emission)
 - [x] Add `components.xml` `distribution.xml` `groups.xml` merge support
 - [x] Add compression and validation files (`.xz` `.sha1sum`)
 - [x] Hook in deltas
 - [x] Throw the entire repo at it and ensure constant time is still _true_
 - [x] Now add the monitor
 - [ ] Profit $$$

**Comment by ikeydoherty** [comment]
Final thought before hitting the pillow:

Might be worth having two worker pools, the low priority delta pool and a high priority reentrant pool for upload processing, to limit the main queue purely to atomic operations

**Comment by ikeydoherty** [comment]
OK I added a dependency based scheduler instead of lame-reentrants. 

We're gonna have to implement our own delta handling as it turns out `eopkg` will not allow us to produce `.delta.eopkg` in parallel ..

**Comment by ikeydoherty** [comment]
done. yet to profit. also, stop talking to yourself, it's weird.
